### ISSUE TITLE
tests_ckm fails to compile with clang

### ISSUE THREAD

**Issue by wkotlarski** [FlexibleSUSY/FlexibleSUSY#130]
```
Building test/test_ckm.o
In file included from test/test_ckm.cpp:12:
In file included from src/linalg2.hpp:30:
In file included from /usr/include/eigen3/Eigen/Eigenvalues:19:
In file included from /usr/include/eigen3/Eigen/Geometry:50:
/usr/include/eigen3/Eigen/src/Geometry/Umeyama.h:140:12: error: read-only variable is not assignable
    S(m-1) = -1;
    ~~~~~~ ^
1 error generated.
make: *** [Makefile:242: test/test_ckm.o] Error 1
```
@uukhas confirms this error. On top of that he also gets some other error. I'm using clang version 9.0.0.

**Comment by Expander** [comment]
Many thanks for raising this issue!

Can one reproduce this error also with an older clang version, such as clang 7.0?

**Comment by Expander** [comment]
fixed in commit 471580233
